### ISSUE TITLE
Credhub Integration Bug in 3.9.0

### ISSUE THREAD

**Issue by scnewma** [concourse/concourse#2034]
# Bug Report

* Concourse version: 3.9.0
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: vSphere
* Did this used to work? Yes (3.8.0)

Credhub integration is not working in `3.9.0`.  It looks like a refactor (9a105c1f59719616af1a41142443b2cdc22fcab4) was done to the `atc_ctl.erb` for updates to Credhub CA Cert.  This moved the `credhub.client_id` and `credhub.client_secret` to `credhub.tls.client_id` and `credhub.tls.client_secret` respectively.  The atc spec is still referring to them as `credhub.client_id` and `credhub.client_secret`.

**Comment by jama-pivotal** [comment]
@scnewma what version of credhub are you using with 3.9.0?

**Comment by scnewma** [comment]
@jama-pivotal 1.6.5

**Comment by rahulkj** [comment]
Exact same issue and also getting the error `provided ca certs are invalid` while upgrading to 3.9.0

**Comment by vito** [comment]
Sorry about that, those properties are indeed borked. The template should have actually been referring to them as `credhub.client_id` and `credhub.client_secret`. The `credhub.tls.*` stuff is different; that's for configuring client *cert* based auth.

Ironically this was all refactored to improve testability, and we're now testing CredHub integration in our pipeline, but we're testing it with *cert* based auth as deploying a UAA and configuring the client ID/secret based auth is too heavy-handed and annoying for our pipeline. So in adding test coverage we broke the one use case people were using before. :( Sorry. This will be out in 3.9.1.

@rahulkj That may be a different behavior. Prior to 3.9.0 the CA cert field was actually completely ignored, and I believe the ATC would just always skip verification. This was a bug that we fixed. So it may be that the CA cert that you've configured this whole time is actually invalid or malformed.

@scnewma FWIW, as of 3.9.0 we're testing with CredHub 1.7.0. I'm not sure if 1.6.5 will work with 3.9.0. (It might - just pointing out we haven't tested it, as our pipeline has moved forward.)
